ts/features/wallet/component/PagoPaPaymentCapability.tsx

```diff
@@ -0,0 +1,117 @@
+import { fromNullable, isNone } from "fp-ts/lib/Option";
+import { Badge, View } from "native-base";
+import * as React from "react";
+import { StyleSheet } from "react-native";
+import { H4 } from "../../../components/core/typography/H4";
+import { H5 } from "../../../components/core/typography/H5";
+import { IOColors } from "../../../components/core/variables/IOColors";
+import { IOStyles } from "../../../components/core/variables/IOStyles";
+import I18n from "../../../i18n";
+import { CreditCardType, PaymentMethod } from "../../../types/pagopa";
+
+const styles = StyleSheet.create({
+  row: {
+    flexDirection: "row",
+    justifyContent: "space-between"
+  },
+  left: {
+    ...IOStyles.flex,
+    paddingRight: 8
+  },
+  badgeInfo: {
+    borderWidth: 1,
+    borderStyle: "solid",
+    height: 25,
+    flexDirection: "row"
+  },
+  badgeAvailable: {
+    backgroundColor: IOColors.blue,
+    borderColor: IOColors.blue,
+    width: 65
+  },
+  badgeArriving: {
+    backgroundColor: IOColors.white,
+    borderColor: IOColors.blue,
+    width: 65
+  },
+  badgeNotAvailable: {
+    backgroundColor: IOColors.white,
+    borderColor: IOColors.blue,
+    width: 130
+  }
+});
+
+type Props = { paymentMethod: PaymentMethod };
+
+type badgeType = "available" | "arriving" | "not_available";
+
+const getLocales = () => ({
+  available: I18n.t("wallet.methods.card.pagoPaCapability.active"),
+  arriving: I18n.t("wallet.methods.card.pagoPaCapability.arriving"),
+  incompatible: I18n.t("wallet.methods.card.pagoPaCapability.incompatible")
+});
+
+const availabilityBadge = (badgeType: badgeType) => {
+  const { available, arriving, incompatible } = getLocales();
+  switch (badgeType) {
+    case "available":
+      return (
+        <Badge style={[styles.badgeInfo, styles.badgeAvailable]}>
+          <H5 color="white">{available}</H5>
+        </Badge>
+      );
+    case "arriving":
+      return (
+        <Badge style={[styles.badgeInfo, styles.badgeArriving]}>
+          <H5 color="blue">{arriving}</H5>
+        </Badge>
+      );
+    case "not_available":
+      return (
+        <Badge style={[styles.badgeNotAvailable, styles.badgeInfo]}>
+          <H5 color="blue">{incompatible}</H5>
+        </Badge>
+      );
+  }
+};
+
+const extractBadgeType = (paymentMethod: PaymentMethod): badgeType => {
+  const badgeType = fromNullable(paymentMethod).map<badgeType>(pM => {
+    switch (pM.kind) {
+      case "CreditCard":
+        return CreditCardType.decode(pM.info.brand).fold(
+          () => "not_available",
+          b => (b === "MAESTRO" ? "arriving" : "available")
+        );
+      case "Satispay":
+        return "arriving";
+      default:
+        return "not_available";
+    }
+  });
+  return isNone(badgeType) ? "not_available" : badgeType.value;
```

Comment:
@fabriziofff I also thought this was the behavior, but I tried with a **Maestro** card with the production server and it returns `pagoPa = true`.

Reply:
Maestro is compatible, but it requires cvv each time you perform a payment—and our current app doesn't support this feature yet.

Reply:
btw this statement could show a credit card with witch is not possibile to pay (ex: brand `VISA` but `pagopa=false`). We have to consider the [pagoPa](https://github.com/pagopa/io-app/pull/2606/files#r544918086) attribute and the MAESTRO brand to say "with this method you can't pay with pagoPA"

So my suggestion is to create an utility function (`canCreditCardPay`) and write some test on it since it could change few times in the next moths

What about this?

```typescript
const brandsBlackList = new Set<CreditCardType>(["MAESTRO"]);
const isSupportedBrand = (
  creditCard: CreditCardPaymentMethod
): boolean =>
  CreditCardType.decode(creditCard.info.brand).fold(
    () => false,
    pm => !brandsBlackList.has(pm)
  );

const extractBadgeType = (paymentMethod: PaymentMethod): badgeType => {
  const badgeType = fromNullable(paymentMethod).map<badgeType>(pM => {
    switch (pM.kind) {
      case "CreditCard":
        return pM.pagoPA
          ? isSupportedBrand(pM)
            ? "available"
            : "arriving"
          : "not_available";
      case "Satispay":
        return "arriving";
      default:
        return "not_available";
    }
  });
  return badgeType.getOrElse("not_available");
};
```